codex-rs/tui/src/lib.rs

```diff
@@ -495,7 +495,12 @@ fn determine_repo_trust_state(
         // if the current cwd project is trusted and no config has been set
         // skip the trust flow and set the approval policy and sandbox mode
         config.approval_policy = AskForApproval::OnRequest;
-        config.sandbox_policy = SandboxPolicy::new_workspace_write_policy();
+        // Respect workspace‑write settings from config.toml
+        config.sandbox_policy = if config_toml.sandbox_workspace_write.is_some() {
```

Comment:
Why is this happening in the TUI crate? Shouldn't the `Config` be finalized before we get here? Or even if it isn't, should this live in `core`?